src/Nethermind/Nethermind.Evm/Tracing/GethStyle/GethLikeTxTracer.cs

```diff
@@ -222,26 +189,34 @@ public void ReportAccess(IReadOnlySet<Address> accessedAddresses, IReadOnlySet<S
 
         public void SetOperationStack(List<string> stackTrace)
         {
-            _traceEntry.Stack = stackTrace;
+            CurrentTraceEntry.Stack = stackTrace;
         }
 
         public void ReportStackPush(in ReadOnlySpan<byte> stackItem)
         {
         }
 
-        public void SetOperationMemory(List<string> memoryTrace)
+        public void SetOperationMemory(IEnumerable<string> memoryTrace)
         {
-            _traceEntry.Memory = memoryTrace;
+            if (IsTracingFullMemory)
+                CurrentTraceEntry.Memory = memoryTrace.ToList();
```

Comment:
maybe keep it as an IEnumerable for FileTrace and just write it to file?

Reply:
I think that could be done with a `Union`? But for if multiple of them are being created before serializing, we do need Lists.